meetings/2020-11/nov-16.md

```diff
@@ -0,0 +1,678 @@
+
+MM: I feel strongly they should be bundled together. The way to remove the setting behavior to remove the accessor property.
+
+JHD: Does anyone else have thoughts on that? if not, I think we'll just go with bundling the two of them together.
+
+MF: My preference aligns with Mark.
+
+#### Conclusion/Resolution
+`__proto__` getter and setter will be bundled
+## Give %TypedArray% methods explicit algorithms
+
+Presenter: Shu-yu Guo (SYG)
+
+- [pr](https://github.com/tc39/ecma262/pull/2221)
+- [slides](https://docs.google.com/presentation/d/1Um10s6wNWwRiUgxMu8Ur7jrLJPh9LZAVNdDlhUIF9bk/edit)
+
+SYG: So this is another thing where in Ross's quest to get the spec to reflect reality about typed arrays, we found some other parts that were under specified and have some engine disagreement. So this is one of those follow-on PRs I'm presenting for Ross. So currently on typedArray.prototype for the different TypedArrays, the first bullet point lists [on the slide] all the things that do not have algorithm steps. They only have prose. The only two methods that have algorithm steps are map and filter. So what the pr does is it tries to give everything algorithm steps, but we ran into some issues. So first of all what the prose currently says is the following three things. I'm paraphrasing of course, but they all basically say to implement the same thing as the array dot prototype counterparts except when you would read that length property instead read the internal slot array length on the typed array and at the start of every method call validate typed array basically all valid a type. The right does is it throws if the typed array is detached at the beginning of the method call? And what this PR does is then tries to follow the spirit of what the process would be. So it first copies the spec text word for word step for step from the array dot prototype, then inserts a valid typedarray call at the start, then substitutes all the reads of the length property with reads of the array length internal slot. And this is highlighted because this is the part that has disagreement and is under-specified and in red. The has property checks because type of a stand-up holes and implementations don't call hasProperty. Notably the prose doesn't say anything about what you should do about the hasProperty checks. So the implementations kind of interpreted some way and did something. Luckily only three methods care about the HasProperty checks. So steps 1 to 3 here is sufficient to fully define in a compatible way for all the methods except these three methods: includes indexOf and lastIndexOf. Once a typed array type to raise the underlying buffer is detached, type 3 behaves as if they have no index to come properties. So all calls to the hasProperty, the hasProperty internal method will return false for its properties.
```

Comment:
```suggestion
SYG: So this is another thing where in Ross's quest to get the spec to reflect reality about typed arrays, we found some other parts that were under specified and have some engine disagreement. So this is one of those follow-on PRs I'm presenting for Ross. So currently on TypedArray.prototype for the different TypedArrays, the first bullet point lists [on the slide] all the things that do not have algorithm steps. They only have prose. The only two methods that have algorithm steps are map and filter. So what the PR does is it tries to give everything algorithm steps, but we ran into some issues. So first of all what the prose currently says is the following three things. I'm paraphrasing of course, but they all basically say to implement the same thing as the Array.prototype counterparts except when you would read that length property, instead read the internal slot [[ArrayLength]] on the typed array, and at the start of every method call validate typed array basically all valid a type. All it does is it throws if the typed array is detached at the beginning of the method call. And what this PR does is then tries to follow the spirit of what the process would be. So it first copies the spec text word for word step for step from the Array.prototype, then inserts ValidateTypedArray call at the start, then substitutes all the reads of the length property with reads of the [[ArrayLength]] internal slot. And this is highlighted because this is the part that has disagreement and is under-specified and in red. The [[HasProperty]] checks because TypedArray doesn't have holes and implementations don't call [[HasProperty]]. Notably the prose doesn't say anything about what you should do about the [[HasProperty]] checks. So the implementations kind of interpreted some way and did something. Luckily only three methods care about the HasProperty checks. So steps 1 to 3 here is sufficient to fully define in a compatible way for all the methods except these three methods: includes indexOf and lastIndexOf. Once a typed array type to raise the underlying buffer is detached, type 3 behaves as if they have no index to come properties. So all calls to the hasProperty, the [[HasProperty]] internal method will return false for its properties.
```